.pipelines/npm/npm-scale-test.yaml

```diff
@@ -0,0 +1,418 @@
+              -g $CLUSTER_NAME \
+              -n nodepool1 \
+              --node-taints CriticalAddonsOnly=true:NoSchedule
+
+            echo "Adding Windows nodepool to $CLUSTER_NAME"
+            az aks nodepool add \
+                --resource-group $CLUSTER_NAME \
+                --cluster-name $CLUSTER_NAME \
+                --name awin22 \
+                --os-type Windows \
+                --os-sku Windows2022 \
+                --node-vm-size Standard_D4s_v3 \
+                --node-count 1 \
+                --max-pods 100
+
+            echo "Getting credentials to $CLUSTER_NAME"
+            az aks get-credentials -g $CLUSTER_NAME -n $CLUSTER_NAME --overwrite-existing --file ./kubeconfig
+            mkdir -p ~/.kube/
+            cp ./kubeconfig ~/.kube/config
+
+      - task: AzureCLI@2
+        displayName: "Deploy NPM to Test Cluster"
+        inputs:
+          azureSubscription: $(BUILD_VALIDATIONS_SERVICE_CONNECTION)
+          scriptType: "bash"
+          scriptLocation: "inlineScript"
+          failOnStderr: true
+          condition: succeeded()
+          inlineScript: |
+            set -e
+
+            # deploy azure-npm
+            cp $(Pipeline.Workspace)/s/npm/examples/windows/azure-npm.yaml azure-npm.yaml
+            # set higher memory limit
+            sed -i 's/memory: 300Mi/memory: 1000Mi/g' azure-npm.yaml
+            kubectl apply -f azure-npm.yaml
+
+            # swap azure-npm image with one built during run
+            kubectl set image daemonset/azure-npm-win -n kube-system azure-npm=$IMAGE_REGISTRY/azure-npm:windows-amd64-ltsc2022-$(TAG)
+
+            sleep 5s
+            echo "waiting for NPM to start running..."
+            kubectl wait --for=condition=Ready pod -l k8s-app=azure-npm -n kube-system --timeout=20m
+            echo "sleep 3m to let NPM restart in case of bootup failure due to HNS errors"
+            sleep 3m
+
+            kubectl get po -n kube-system -owide -A
+
+            echo "labeling Windows nodes for scale test"
+            kubectl get node -o wide | grep "Windows Server 2022 Datacenter" | awk '{print $1}' | xargs -n 1 -I {} kubectl label node {} scale-test=true connectivity-test=true
+
+            export CLUSTER_NAME=$(RESOURCE_GROUP)-$(PROFILE)
+            echo "Showing cluster status for $CLUSTER_NAME"
+            FQDN=`az aks show -n $CLUSTER_NAME -g $CLUSTER_NAME --query fqdn -o tsv`
+            echo "##vso[task.setvariable variable=FQDN]$FQDN"
+
+      - task: AzureCLI@2
+        displayName: "Scale Up Large"
+        inputs:
+          azureSubscription: $(BUILD_VALIDATIONS_SERVICE_CONNECTION)
+          scriptType: "bash"
+          scriptLocation: "inlineScript"
+          failOnStderr: true
+          condition: succeeded()
+          inlineScript: |
+            set -e
+            mkdir -p $(System.DefaultWorkingDirectory)/$(RESOURCE_GROUP)-$(PROFILE)
+            ./kwok --kubeconfig ~/.kube/config \
```

Comment:
this is pretty useful for other scenarios, maybe after this add a make recipe to attach kwok nodes